biothings/web/handlers/base.py

```diff
@@ -52,9 +52,22 @@ class BaseAPIHandler(BaseHandler, AnalyticsMixin):
         }
     }
     format = 'json'
-    cache = 604800  # 7 days
+    cache = None
 
-    def initialize(self):
+    def initialize(self,  cache=None):
+        cache_value = self.biothings.config.DEFAULT_CACHE_MAX_AGE
+        if self.cache is not None:
+            cache_value = self.cache
+        if cache is not None:
+            cache_value = cache
+
+        if isinstance(cache_value, int):
+            # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
+            # to disable caching for a handler, set cls.cache to 0 or
+            # run self.clear_header('Cache-Control') in an HTTP method
+            # or set cache value on the config file:
+            # r"/api/query/?", "biothings.web.handlers.QueryHandler", {"biothing_type": "schema", "cache": 0}),
+            self.set_header("Cache-Control", f"max-age={cache_value}, public")
```

Comment:
@remoteeng00 also probably useful to put this string `"max-age={cache_value}, public"` a class-level property, like `self.cache_control_template = "max-age={%(cache)s}, public"`. This will allow subclass to override to something else, e.g. set it to`"max-age={%(cache)s}, private"`.

Reply:
Is it necessary if we have the `set_cache_header` func?

Reply:
probably just for a convenient alternative of overriding `set_cache_header` method, we can set it as a variable instead of hard-coded at the class level:

```
class BaseAPIHandler(BaseHandler, AnalyticsMixin):
    ...
    cache = None
    cache_control_template = "max-age={%(cache)s}, public"
```

That's it. No need to make it a global config, or handler parameter.

Reply:
okay, it's added.